GitCommands/Utils/EnvUtils.cs

```diff
@@ -19,6 +20,21 @@ public static bool RunningOnWindows()
             }
         }
 
+        public static bool RunningOnWindowsWithMainWindow()
+        {
+            if (!RunningOnWindows())
+            {
+                return false;
+            }
+
+            if (!(Process.GetCurrentProcess() is { } currentProcess))
+            {
+                return false;
+            }
+
+            return currentProcess.MainWindowHandle != IntPtr.Zero;
```

Comment:
(Strange syntax for C# amateurs.)

Reply:
It's a bit strange, but from the meetings I've seen to date it looks like this is going to stay as the "read to local and check for non-null" form.

Reply:
I can change this to the following if you prefer:

```csharp
var currentProcess = Process.GetCurrentProcess();
if (currentProcess is null)
{
  return false;
}
```

Reply:
I haven't seen this syntax until now, but I figured it would mean null-check. 
C# more and more becomes likes a JS... 😝

Reply:
Yes, please. This is much more readable.

Reply:
Done